neural_solution/backend/cluster.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+import sqlite3
+from typing import List
+from neural_solution.backend.utils.utility import synchronized, create_dir
+from neural_solution.utils import logger
+from collections import Counter
+
+class Cluster:
+    def __init__(self, node_lst=[], db_path=None):
+        """Cluster manages all the server resources.
+
+        TaskDb provides atomic operations on managing the resource dict.
+
+        Attributes:
```

Comment:
func/class comments look not correct. pls follow our recommended doxgen comment style.